FIPS approved crypto for NATS server xkey

### Observed behavior

The `xkey` created [here](https://github.com/nats-io/nats-server/blob/main/server/server.go#L726) uses curve25519, which is not FIPS compliant.

If you deploy the NATS server with `GODEBUG=fips140=only`, the following panic occurs:

```
panic: curve25519: internal error: scalarBaseMult was not 32 bytes

goroutine 1 [running]:
golang.org/x/crypto/curve25519.ScalarBaseMult(0x4000159988, 0x55ef14?)
    golang.org/x/crypto@v0.41.0/curve25519/curve25519.go:39 +0x10c
github.com/nats-io/nkeys.(*ckp).PublicKey(0x0?)
    github.com/nats-io/nkeys@v0.4.11/xkeys.go:81 +0x30
github.com/nats-io/nats-server/v2/server.NewServer(0x400019c008)
    github.com/nats-io/nats-server/v2/server/server.go:674 +0xac
main.main()
    github.com/nats-io/nats-server/v2/main.go:118 +0x204
stream closed EOF for mural-system/nats-0 (nats)
```

### Expected behavior

NATS server should either use ECDH over P-256 by default or support configuration options to use that algorithm.

### Server and client version

N/A - referenced code is on main.

### Host environment

_No response_

### Steps to reproduce

Deploy NATS server with `GODEBUG=fips140=only`.

--- a/server/ciphersuites.go
+++ b/server/ciphersuites.go
@@ -14,6 +14,7 @@
 package server
 
 import (
+	"crypto/fips140"
 	"crypto/tls"
 )
 
@@ -52,6 +53,16 @@ var curvePreferenceMap = map[string]tls.CurveID{
 // reorder to default to the highest level of security.  See:
 // https://blog.bracebin.com/achieving-perfect-ssl-labs-score-with-go
 func defaultCurvePreferences() []tls.CurveID {
+	if fips140.Enabled() {
+		// X25519 is not FIPS-approved by itself, but it is when
+		// combined with MLKEM768.
+		return []tls.CurveID{
+			tls.X25519MLKEM768, // post-quantum
+			tls.CurveP256,
+			tls.CurveP384,
+			tls.CurveP521,
+		}
+	}
 	return []tls.CurveID{
 		tls.X25519MLKEM768, // post-quantum
 		tls.X25519,         // faster than P256, arguably more secure

--- a/server/opts.go
+++ b/server/opts.go
@@ -15,6 +15,7 @@ package server
 
 import (
 	"context"
+	"crypto/fips140"
 	"crypto/tls"
 	"crypto/x509"
 	"errors"
@@ -2484,6 +2485,9 @@ func parseJetStreamTPM(v interface{}, opts *Options, errors *[]error) error {
 func setJetStreamEkCipher(opts *Options, mv interface{}, tk token) error {
 	switch strings.ToLower(mv.(string)) {
 	case "chacha", "chachapoly":
+		if fips140.Enabled() {
+			return &configErr{tk, fmt.Sprintf("Cipher type %q cannot be used in FIPS-140 mode", mv)}
+		}
 		opts.JetStreamCipher = ChaCha
 	case "aes":
 		opts.JetStreamCipher = AES
@@ -4378,6 +4382,10 @@ func parseAuthorization(v any, errors, warnings *[]error) (*authorization, error
 			}
 			auth.defaultPermissions = permissions
 		case "auth_callout", "auth_hook":
+			if fips140.Enabled() {
+				*errors = append(*errors, fmt.Errorf("'auth_callout' cannot be configured in FIPS-140 mode"))
+				continue
+			}
 			ac, err := parseAuthCallout(tk, errors)
 			if err != nil {
 				*errors = append(*errors, err)

--- a/server/server.go
+++ b/server/server.go
@@ -16,6 +16,7 @@ package server
 import (
 	"bytes"
 	"context"
+	"crypto/fips140"
 	"crypto/tls"
 	"encoding/json"
 	"errors"
@@ -722,8 +723,12 @@ func NewServer(opts *Options) (*Server, error) {
 	pub, _ := kp.PublicKey()
 
 	// Create an xkey for encrypting messages from this server.
-	xkp, _ := nkeys.CreateCurveKeys()
-	xpub, _ := xkp.PublicKey()
+	var xkp nkeys.KeyPair
+	var xpub string
+	if !fips140.Enabled() {
+		xkp, _ = nkeys.CreateCurveKeys()
+		xpub, _ = xkp.PublicKey()
+	}
 
 	serverName := pub
 	if opts.ServerName != _EMPTY_ {

